pkg/trait/service.go

```diff
@@ -46,7 +46,7 @@ func newServiceTrait() Trait {
 // IsAllowedInProfile overrides default.
 func (t *serviceTrait) IsAllowedInProfile(profile v1.TraitProfile) bool {
 	return profile.Equal(v1.TraitProfileKubernetes) ||
-		profile.Equal(v1.TraitProfileOpenShift)
+		profile.Equal(v1.TraitProfileOpenShift) || profile.Equal(v1.TraitProfileKnative)
```

Comment:
By default `IsAllowedInProfile()` allows every profile, so instead of adding another profile you can simply remove this override.

Reply:
Oh, yes, thanks for noticing this.